### PR TITLE
HARP-10237 Refactor VisibleTileSet

### PR DESCRIPTION
**Pull request by nzjony**
- Cache unique tile key of Tile
- Rename method to improve ease of understanding
- Reorder variable definitions to be closer to their respective usage

Signed-off-by: Jonathan Stichbury <2533428+nzjony@users.noreply.github.com>

### CODE REVIEW

File: @here/harp-mapview/lib/VisibleTileSet.ts [hunk 1]
```diff
@@ -886,76 +895,65 @@ export class VisibleTileSet {
...
+            const { searchDirection } = this.getSearchDirection(dataSource, dataZoomLevel);
+
             // Minor optimization for the fallback search, only check parent tiles once, otherwise
             // the recursive algorithm checks all parent tiles multiple times, the key is the code
             // of the tile that is checked and the value is whether a parent was found or not.
             const checkedTiles = new Map<number, boolean>();
             // Iterate over incomplete (not loaded tiles) and find their parents or children that
             // are in cache that can be rendered temporarily until tile is loaded. Note, we favour
             // falling back to parent tiles rather than children.
-            for (const [tileKeyCode, searchDirection] of incompleteTiles) {
```
Comment (nzjony) [inline @here/harp-mapview/lib/VisibleTileSet.ts:958]: searchDirection is consistent for each tile, so I moved it out.

File: @here/harp-mapview/lib/VisibleTileSet.ts [hunk 2]
```diff
@@ -886,76 +895,65 @@ export class VisibleTileSet {
...
+            // tiles that are incomplete, and for which we search for an alternative
+            // ("incompleteTiles").
             renderListEntry.visibleTiles.forEach(tile => {
-                const tileCode = TileOffsetUtils.getKeyForTileKeyAndOffset(
-                    tile.tileKey,
-                    tile.offset
-                );
                 tile.levelOffset = 0;
                 if (tile.hasGeometry) {
-                    renderedTiles.set(tileCode, tile);
```
Comment (nzjony) [inline @here/harp-mapview/lib/VisibleTileSet.ts:938]: minor optimization / code improvement by computing it only once and caching in the Tile

File: @here/harp-mapview/lib/Tile.ts [hunk 3]
```diff
@@ -387,6 +388,18 @@ export class Tile implements CachedResource {
         return this.m_worldCenter;
     }
 
+    /**
+     * Gets the key to uniquely represent this tile (based on the [[tileKey]] and [[offset]]), note
+     * this key is only unique within the given [[DataSource]], to get a key which is unique across
+     * [[DataSource]]s see [[DataSourceCache.DataSourceCache]].
+     */
+    get uniqueKey(): number {
```
Comment (zbigg) [inline @here/harp-mapview/lib/Tile.ts:396]: Very minor, i think that uniqueKey will be _always_ used for any tile that's actually used, so i believe lazy optimization is not necessarily needed, maybe just create this `uniqueKey` in constructor ?

Comment (nzjony) [reply @here/harp-mapview/lib/Tile.ts:396]: agreed, and it won't change either, will move it.

**Comment by nzjony** [comment]
@harpgl-bot retest this please

File: @here/harp-mapview/lib/Tile.ts (hunk 3 above)
Comment (nzjony) [reply @here/harp-mapview/lib/Tile.ts:397]: of course I was wrong, in order not to pollute the DataSource.getTile method, I didn't add the offset, and the offset is set manually later.

That means however, the uniqueKey needs to be updated when the offset is changed, I have fixed this.

**Comment by nzjony** [comment]
@harpgl-bot retest this please

**Pull request closed without merging**